sdk/mobile-center/src/main/java/com/microsoft/azure/mobile/utils/storage/StorageHelper.java

```diff
@@ -57,8 +57,10 @@
      * @param context The context of the application.
      */
     public static void initialize(Context context) {
-        sContext = context;
-        sSharedPreferences = sContext.getSharedPreferences(PREFERENCES_NAME, Context.MODE_PRIVATE);
+        if (sContext == null) {
```

Comment:
I don't understand why you talk about delay and null pointers.

I introduced that optimization to avoid doing duplicate initializations. I have a case in Updates where I need to initialize storage if MobileCenter not yet started. The code is not yet synchronized as for now we don't do that in 2 separate threads.